src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -1004,7 +1009,32 @@ internal static IEnumerable<EnvDTE.Project> GetDependentEnvDTEProjects(IDictiona
             dependentEnvDTEProjects.Add(dependentEnvDTEProject);
         }
 
-#region IVsSelectionEvents
+        private void NuGetCacheUpdate_After(object sender, EventArgs e)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(() => FireNuGetCahceUpdatedEventAsync(e));
+        }
+
+        private async Task FireNuGetCahceUpdatedEventAsync(EventArgs e)
+        {
+            try
+            {
+                // Await a delay of 100 mSec to batch multiple cache updated events
+                await Task.Delay(100);
+                // Check if the cache is still dirty
+                if (_projectSystemCache.TestResetDirtyFlag())
+                {
+                    // Fire the event only if the cache is dirty
+                    AfterNuGetCacheUpdated?.Invoke(this, e);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex.ToString());
```

Comment:
log the exception at the debug level or to the activity log. Displaying a stack trace to the user and not actually stopping the action seems incorrect.